package.json

```diff
@@ -4,57 +4,39 @@
   "private": true,
   "type": "module",
   "scripts": {
-    "build:clean": "turbo run build:clean",
-    "build": "turbo run build:clean; turbo run build --concurrency=30",
-    "watch": "turbo run watch --concurrency=50 --continue",
-    "gallery": "turbo run dev:gallery",
-    "laboratory": "turbo run dev:laboratory",
-    "demo": "turbo run dev:demo",
-    "examples": "turbo run dev:example",
-    "build:gallery": "turbo run build:clean; turbo run build:gallery",
-    "build:laboratory": "npm run build; turbo run build:laboratory",
-    "build:demo": "turbo run build:clean; turbo run build:demo",
-    "build:examples": "turbo run build:clean; turbo run build:examples",
-    "test": "turbo run test",
+    "build:clean": "pnpm --if-present -r clean",
+    "build": "pnpm --if-present --workspace-concurrency=30 -r build",
+    "watch": "pnpm --if-present --workspace-concurrency=50 -r watch",
+    "gallery": "pnpm --filter=./apps/gallery dev",
+    "laboratory": "pnpm --filter=./apps/laboratory dev",
+    "demo": "pnpm --filter=./apps/gallery dev",
+    "examples": "pnpm --filter=./examples -r dev",
+    "build:packages": "pnpm --filter=./packages/* build",
+    "build:gallery": "pnpm build:packages; pnpm --filter=./apps/gallery build",
+    "build:laboratory": "pnpm build:packages; pnpm --filter=./apps/laboratory build",
+    "build:demo": "pnpm build:packages; pnpm --filter=./apps/demo build",
+    "build:examples": "pnpm build:packages; pnpm --filter=./examples -r build",
+    "test": "pnpm -r test --if-present",
     "coverage:merge": "node ./scripts/coverage-merge.js",
     "coverage:summary": "node ./scripts/coverage-summary.js",
-    "typecheck": "turbo run typecheck",
-    "lint": "turbo run lint",
+    "typecheck": "pnpm --if-present -r typecheck",
+    "lint": "pnpm -r lint",
```

Comment:
[-r parameter](https://pnpm.io/cli/remove#--recursive--r) not really clear for me. Do we have to use it? It looks like we need to be careful about, could you explain why we used it?

Reply:
`--recursive` flag is used to run a script in every workspace package.

Reply:
Ah yes, the way they described is quite quite different just why:
> When used inside a [workspace](https://pnpm.io/workspaces), removes a dependency (or dependencies) from every workspace package.